src/Bicep.LangServer/BicepCompilationManager.cs

```diff
@@ -273,6 +290,78 @@ private ImmutableArray<ISourceFile> CloseCompilationInternal(DocumentUri documen
             }
         }
 
+        private void SendTelemetryIfLinterRuleWasDisabledInBicepConfig(RootConfiguration prevConfiguration, RootConfiguration curConfiguration)
+        {
+            foreach (var telemetryEvent in GetTelemetryEventsForBicepConfigChange(prevConfiguration, curConfiguration))
+            {
+                TelemetryProvider.PostEvent(telemetryEvent);
+            }
+        }
+
+        public IEnumerable<BicepTelemetryEvent> GetTelemetryEventsForBicepConfigChange(RootConfiguration prevConfiguration, RootConfiguration curConfiguration)
+        {
+            var linterEnabledSetting = "core.enabled";
+            bool prevLinterEnabledSettingValue = prevConfiguration.Analyzers.GetValue(linterEnabledSetting, true);
+            bool curLinterEnabledSettingValue = curConfiguration.Analyzers.GetValue(linterEnabledSetting, true);
+
+            if (!prevLinterEnabledSettingValue && !curLinterEnabledSettingValue)
+            {
+                return Enumerable.Empty<BicepTelemetryEvent>();
+            }
+
+            List<BicepTelemetryEvent> telemetryEvents = new();
+
+            if (prevLinterEnabledSettingValue != curLinterEnabledSettingValue)
+            {
+                var telemetryEvent = BicepTelemetryEvent.CreateOverallLinterStateChangeInBicepConfig(prevLinterEnabledSettingValue.ToString().ToLowerInvariant(), curLinterEnabledSettingValue.ToString().ToLowerInvariant());
+                telemetryEvents.Add(telemetryEvent);
+            }
+            else
+            {
+                foreach (var kvp in LinterRules)
+                {
+                    string prevLinterRuleDiagnosticLevelValue = prevConfiguration.Analyzers.GetValue(kvp.Value, "warning");
+                    string curLinterRuleDiagnosticLevelValue = curConfiguration.Analyzers.GetValue(kvp.Value, "warning");
+
+                    if (prevLinterRuleDiagnosticLevelValue != curLinterRuleDiagnosticLevelValue)
+                    {
+                        var telemetryEvent = BicepTelemetryEvent.CreateDisableRuleInBicepConfig(kvp.Key, prevLinterRuleDiagnosticLevelValue, curLinterRuleDiagnosticLevelValue);
```

Comment:
>CreateDisableRuleInBicepConfig [](http://example.com/codeflow?start=65&length=30)

This doesn't represent disabling but a state change. Can you rename the event to match?

Reply:
Ah, thank you!

Updated here:
https://github.com/Azure/bicep/commit/d6cddc31b4b5d02163ccf4e0f69b581d8ffece9a